### PR TITLE
Some usability improvements to the main GUI

### LINKED ISSUE #2245: Why is DontUseNativeDialog specified in project creation
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Why is DontUseNativeDialog specified in https://github.com/DeepLabCut/DeepLabCut/blob/c2469aba67a9f732a678ba1e47cdf7171b8b0a49/deeplabcut/gui/tabs/create_project.py#L112

It seems like we would want to use native dialogs if we can? It seems to have been added in the latest big update to PySide6
https://github.com/DeepLabCut/DeepLabCut/commit/993a64021762c3b5e2c639ba942f445f66c25a2f

I understand there are still some challenges with PySide6 so maybe i'm missing something

### Operating System

linux + conda-forge

### DeepLabCut version

2.3.4

### DeepLabCut mode

single animal

### Device type

cpu

### Steps To Reproduce

Create a new project
Add videos

Notice that the native dialog isn't use on linux

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jeylau**
Tested by @n-poulsen, @Timokleia and myself w/ napari==0.4.18 and PySide6<6.3.2.

Closes #2245

### CODE REVIEW

File: deeplabcut/gui/tabs/create_project.py [hunk 1]
```diff
@@ -48,16 +51,24 @@ def lay_out_user_frame(self):
 
         proj_label = QtWidgets.QLabel("Project:", user_frame)
         self.proj_line = QtWidgets.QLineEdit(self.proj_default, user_frame)
+        self.proj_line.setPlaceholderText("my project's name")
```
Comment (n-poulsen) [inline deeplabcut/gui/tabs/create_project.py:54]: I'd find it more intuitive that I needed to fill out the text if the placeholder had a different font color (e.g. a light grey, or simply give it reduced opacity)

Comment (jeylau) [reply deeplabcut/gui/tabs/create_project.py:54]: It should; can you try to update PySide6? If that solves it we can probably remove the pin.

Comment (MMathisLab) [reply deeplabcut/gui/tabs/create_project.py:54]: bump @n-poulsen

Comment (n-poulsen) [reply deeplabcut/gui/tabs/create_project.py:54]: upgrading PySide6 from 6.3.1 to 6.5.1.1 fixed the issue 👍

Comment (n-poulsen) [reply deeplabcut/gui/tabs/create_project.py:54]: Though now when using PySide6 6.5.1.1 I can no longer label frames:

```
Traceback (most recent call last):
  File ".../deeplabcut/gui/tabs/label_frames.py", line 59, in label_frames
    _ = launch_napari(folder)
  File ".../deeplabcut/gui/widgets.py", line 38, in launch_napari
    viewer = napari.Viewer()
  File ".../lib/python3.9/site-packages/napari/viewer.py", line 67, in __init__
    self._window = Window(self, show=show)
  File ".../lib/python3.9/site-packages/napari/_qt/qt_main_window.py", line 463, in __init__
    self._qt_window = _QtMainWindow(viewer)
  File ".../lib/python3.9/site-packages/napari/_qt/qt_main_window.py", line 88, in __init__
    self._qt_viewer = QtViewer(viewer, show_welcome_screen=True)
  File ".../lib/python3.9/site-packages/napari/_qt/qt_viewer.py", line 217, in __init__
    self._create_canvas()
  File ".../lib/python3.9/site-packages/napari/_qt/qt_viewer.py", line 401, in _create_canvas
    self.canvas = VispyCanvas(
  File ".../lib/python3.9/site-packages/napari/_vispy/canvas.py", line 33, in __init__
    super().__init__(*args, **kwargs)
...
  File ".../lib/python3.9/site-packages/vispy/app/backends/_qt.py", line 283, in _set_config
    glformat.setSwapBehavior(glformat.DoubleBuffer if c['double_buffer']
AttributeError: 'PySide6.QtGui.QSurfaceFormat' object has no attribute 'DoubleBuffer'
```

Comment (jeylau) [reply deeplabcut/gui/tabs/create_project.py:54]: That was my fear... napari is about to release 0.4.18, I guess they will have addressed most small bugs related to pyside/pyqt6.

Comment (MMathisLab) [reply deeplabcut/gui/tabs/create_project.py:54]: any updates on napari release?

Comment (jeylau) [reply deeplabcut/gui/tabs/create_project.py:54]: no problem on my side with the recent napari 0.4.18 update and PySide6 6.5.1.1; looks like they squashed these small bugs.

Comment (MMathisLab) [reply deeplabcut/gui/tabs/create_project.py:54]: excellent! can me marked as non-draft now?

Comment (jeylau) [reply deeplabcut/gui/tabs/create_project.py:54]: Will do after next napari-deeplabcut release

**Comment by n-poulsen** [comment]
One issue I have with this release (using PySide6 6.5.1.1 and napari 0.4.18) is selecting videos for a project. When I click on a folder containing videos both are selected at first, but when I try to use the toggles to select only one I get into inconsistent states, with pictograms I don't quite understand.

Initial state
initial_state

Inconsistent state 1:
incosistent_state
2-5e3fbbe56b35)

Inconsistent state 2:
inconsistent_state_2

Are you able to replicate these issues?

**Comment by jeylau** [comment]
Yes, good catch! Updating pyside6 broke the `DragDropListView`, let me look at it, there should be an easy fix.

**Comment by n-poulsen** [comment]
This is more of an issue for `napari-deeplabcut` or `napari`, but with `PySide6-6.3.1` I don't have the placeholder with low alpha (as mentioned above), and with `PySide6-6.5.1` `napari` fails to load with error:

```
Traceback (most recent call last):
  File "/miniconda/envs/dlc/bin/napari", line 8, in <module>
    sys.exit(main())
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/__main__.py", line 564, in main
    _run()
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/__main__.py", line 326, in _run
    viewer = Viewer()
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/viewer.py", line 67, in __init__
    self._window = Window(self, show=show)
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/_qt/qt_main_window.py", line 577, in __init__
    self._qt_viewer.dockLayerList, tabify=False, menu=self.window_menu
  File"/miniconda/envs/dlc/lib/python3.10/site-packages/napari/_qt/qt_viewer.py", line 356, in dockLayerList
    layerListLayout.addWidget(self.layers)
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/_qt/qt_viewer.py", line 334, in layers
    self._layers = QtLayerList(self.viewer.layers)
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/_qt/containers/qt_layer_list.py", line 46, in __init__
    super().__init__(root, parent)
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/_qt/containers/qt_list_view.py", line 43, in __init__
    self.setRoot(root)
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/_qt/containers/_base_item_view.py", line 87, in setRoot
    self.setModel(create_model(root, self))
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/_qt/containers/_factory.py", line 72, in create_model
    return QtLayerListModel(obj, parent=parent)
  File "/miniconda/envs/dlc/lib/python3.10/site-packages/napari/_qt/containers/_base_item_model.py", line 80, in __init__
    super().__init__(parent=parent)
TypeError: object.__init__() takes exactly one argument (the instance to initialize)
```

My `napari --info` output is:
```
napari: 0.4.18
Platform: macOS-13.3.1-arm64-arm-64bit
System: MacOS 13.3.1
Python: 3.10.12 (main, Jul  5 2023, 15:02:25) [Clang 14.0.6 ]
Qt: 6.5.1
PySide6: 
NumPy: 1.23.5
SciPy: 1.10.1
Dask: 2022.01.1
VisPy: 0.12.2
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.8
app-model: 0.1.4
npe2: 0.7.0

OpenGL:
  - GL version:  2.1 Metal - 83.1
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1470x956, scale 2.0
  - screen 2: resolution 2560x1440, scale 1.0

Settings path:
  - /......6f4de/settings.yaml
Plugins:
  - napari: 0.4.18 (77 contributions)
  - napari-console: 0.0.8 (0 contributions)
  - napari-deeplabcut: 0.1.3.dev28+gc5e30c0 (16 contributions)
  - napari-svg: 0.1.10 (2 contributions)
```

**Comment by n-poulsen** [comment]
The issue is solved for me by installing `PyQt6` using `pip install PyQt6`. The output of `napari --info` is then:
```
WARNING: pyqt6 already imported, cannot switch to pyside6
WARNING:vispy:pyqt6 already imported, cannot switch to pyside6
napari: 0.4.18
Platform: macOS-13.3.1-arm64-arm-64bit
System: MacOS 13.3.1
Python: 3.10.12 (main, Jul  5 2023, 15:02:25) [Clang 14.0.6 ]
Qt: 6.5.1
PyQt6: 
NumPy: 1.23.5
SciPy: 1.10.1
Dask: 2023.7.0
VisPy: 0.12.2
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.8
app-model: 0.2.0
npe2: 0.7.0

OpenGL:
  - GL version:  2.1 Metal - 83.1
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1470x956, scale 2.0
  - screen 2: resolution 2560x1440, scale 1.0

Settings path:
  - /Users/niels/Library/Application Support/napari/upa_84ffcbcf74a467a4b2c47af3c3eded2cf816f4de/settings.yaml
Plugins:
  - napari: 0.4.18 (77 contributions)
  - napari-console: 0.0.8 (0 contributions)
  - napari-deeplabcut: 0.1.3.dev29+gbd53843 (16 contributions)
  - napari-svg: 0.1.10 (2 contributions)
```

From the two warnings at the top, it seems that PyQt6 has priority instead of PySide6 and is used. I'm not sure if this will have any side effects when using the labeling GUI.

EDIT:
This works when starting `napari` from the command line, but I continue having the issue when launching the `napari` GUI from the DeepLabCut GUI (clicking on "Label Frames"). So this does not fix it.

EDIT 2:
Some more information is available in the corresponding [napari issue](https://github.com/napari/napari/issues/6061). Using `PySide6==6.4.2` works for me (I get the correct placeholder formatting, and can label images using Napari).

**Comment by n-poulsen** [comment]
I've tested this PR with `napari-0.4.18`, `PySide6-6.3.1` and the latest `napari-deeplabcut` (created a project, extracted & labeled frames, trained a model, evaluated, analyzed a video).

Everything worked as expected, so I think `napari-0.4.18` + `PySide6-6.3.1` is the setup to go with. Even though there's no styling applied to the placeholder text, I think that's really not a blocker.

**Review by MMathisLab — APPROVED** [review]
I didn't not personally test, but lgtm! thanks @jeylau

**Review by n-poulsen — APPROVED** [review]

**Pull request merged**